src/components/AssistantTextProcessingForm.vue

```diff
@@ -365,27 +378,38 @@ export default {
 					} else {
 						this.parseTextFileInputs(taskType)
 					}
-					// add placeholders
-					taskTypes.forEach(tt => {
-						if (tt.id === TEXT2TEXT_TASK_TYPE_ID && tt.inputShape.input) {
-							tt.inputShape.input.placeholder = t('assistant', 'Generate a first draft for a blog post about privacy')
-						} else if (tt.id === 'context_chat:context_chat' && tt.inputShape.prompt) {
-							tt.inputShape.prompt.placeholder = t('assistant', 'What is the venue for the team retreat this quarter?')
-						} else if (tt.id === 'core:text2text:summary' && tt.inputShape.input) {
-							tt.inputShape.input.placeholder = t('assistant', 'Type or paste the text to summarize')
-						} else if (tt.id === 'core:text2text:headline' && tt.inputShape.input) {
-							tt.inputShape.input.placeholder = t('assistant', 'Type or paste the text to generate a headline for')
-						} else if (tt.id === 'core:text2text:topics' && tt.inputShape.input) {
-							tt.inputShape.input.placeholder = t('assistant', 'Type or paste the text to extract the topics from')
-						} else if (tt.id === 'core:text2image' && tt.inputShape.input && tt.inputShape.numberOfImages) {
-							tt.inputShape.input.placeholder = t('assistant', 'landscape trees forest peaceful')
-							tt.inputShape.numberOfImages.placeholder = t('assistant', 'a number')
-						} else if (tt.id === 'core:contextwrite' && tt.inputShape.source_input && tt.inputShape.style_input) {
-							tt.inputShape.style_input.placeholder = t('assistant', 'Shakespeare or an example of the style')
-							tt.inputShape.source_input.placeholder = t('assistant', 'A description of what you need or some original content')
-						}
-					})
-					this.taskTypes = taskTypes
+
+					this.getLastTargetLanguage()
```

Comment:
It would probably work to call this asynchronously and change the task type (in `this.taskTypes`) once you get the answer.

It would mean keeping the previous code (`taskTypes.forEach` and then `this.taskTypes = taskTypes`) and then doing the `this.getLastTargetLanguage().then(r => // find and change the task type in this.taskTypes)`

This would solve the delay issue as the `taskTypes` data prop would be set as soon as possible. It would not be slowed down by `this.getLastTargetLanguage()`.